stumpy/cache.py

```diff
@@ -117,3 +145,72 @@ def _get_cache():
     site_pkg_dir = site.getsitepackages()[0]
     numba_cache_dir = site_pkg_dir + "/stumpy/__pycache__"
     return [f.name for f in pathlib.Path(numba_cache_dir).glob("*nb*") if f.is_file()]
+
+
+def _recompile():
+    """
+    Recompile all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    If the `numba` cache is enabled, this results in saving (and/or overwriting)
+    the cached numba functions to disk.
+    """
+    warnings.warn(CACHE_WARNING)
+    for module_name, func_name in get_njit_funcs():
+        module = importlib.import_module(f".{module_name}", package="stumpy")
+        func = getattr(module, func_name)
+        func.recompile()
+
+    return
+
+
+def _save():
+    """
+    Save all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
+    warnings.filterwarnings("once")
+    warnings.warn(CACHE_WARNING)
+    _enable()
+    _recompile()
+    warnings.filterwarnings("default")
```

Comment:
What is this doing?

Reply:
Since I was changing the behaviour of warnings before this line by doing `warnings.filterwarnings("once")`, I thought I need to add this line to reset it. Because changing the behaviour of warning in function can change it globally if I am not mistaken. Maybe I should have used `warnings.resetwarnings()`